vendors/infineon/secure_elements/pkcs11/iot_pkcs11_trustx.c

```diff
@@ -0,0 +1,3122 @@
+{
+    CK_OBJECT_HANDLE xHandle = eInvalidHandle;
+
+    /* Translate from the PKCS#11 label to local storage file name. */
+    if( 0 == memcmp( pLabel,
+                     &pkcs11configLABEL_DEVICE_CERTIFICATE_FOR_TLS,
+                     sizeof( pkcs11configLABEL_DEVICE_CERTIFICATE_FOR_TLS ) ) )
+    {
+        xHandle = eAwsDeviceCertificate;
+    }
+    else if( 0 == memcmp( pLabel,
+                          &pkcs11configLABEL_DEVICE_PRIVATE_KEY_FOR_TLS,
+                          sizeof( pkcs11configLABEL_DEVICE_PRIVATE_KEY_FOR_TLS ) ) )
+    {
+        /* This operation isn't supported for the OPTIGA(TM) Trust X due to a security considerations
+         * You can only generate a keypair and export a private component if you like */
+    	/* We do assign a handle though, as the AWS can#t handle the lables without having a handle*/
+    	xHandle = eAwsDevicePrivateKey;
+    }
+    else if( 0 == memcmp( pLabel,
+                          &pkcs11configLABEL_DEVICE_PUBLIC_KEY_FOR_TLS,
+                          sizeof( pkcs11configLABEL_DEVICE_PUBLIC_KEY_FOR_TLS ) ) )
+    {
+        xHandle = eAwsDevicePublicKey;
+    }
+    else if( 0 == memcmp( pLabel,
+                          &pkcs11configLABEL_CODE_VERIFICATION_KEY,
+                          sizeof( pkcs11configLABEL_CODE_VERIFICATION_KEY ) ) )
+    {
+        xHandle = eAwsCodeSigningKey;
+    }
+
+    return xHandle;
+}
+
+/*-----------------------------------------------------------*/
+
+/**
+ * @brief Gets the value of an object in storage, by handle.
+ *
+ * Port-specific file access for cryptographic information.
+ *
+ * This call dynamically allocates the buffer which object value
+ * data is copied into.  PKCS11_PAL_GetObjectValueCleanup()
+ * should be called after each use to free the dynamically allocated
+ * buffer.
+ *
+ * @sa PKCS11_PAL_GetObjectValueCleanup
+ *
+ * @param[in] pcFileName    The name of the file to be read.
+ * @param[out] ppucData     Pointer to buffer for file data.
+ * @param[out] pulDataSize  Size (in bytes) of data located in file.
+ * @param[out] pIsPrivate   Boolean indicating if value is private (CK_TRUE)
+ *                          or exportable (CK_FALSE)
+ *
+ * @return CKR_OK if operation was successful.  CKR_KEY_HANDLE_INVALID if
+ * no such object handle was found, CKR_DEVICE_MEMORY if memory for
+ * buffer could not be allocated, CKR_FUNCTION_FAILED for device driver
+ * error.
+ */
+BaseType_t PKCS11_PAL_GetObjectValue( CK_OBJECT_HANDLE xHandle,
+                                 uint8_t ** ppucData,
+                                 uint32_t * pulDataSize,
+                                 CK_BBOOL * pIsPrivate )
+{
+	BaseType_t           ulReturn = CKR_OK;
+    optiga_lib_status_t  xReturn;
+    long                 lOptigaOid = 0;
+    char*                xEnd = NULL;
+    uint8_t              xOffset = 0;
+
+    *pIsPrivate = CK_FALSE;
+
+    *ppucData = pvPortMalloc( 1200 );
```

Comment:
Could a specific error code be returned if this malloc fails?

Regarding 1200 bytes, could this be moved to a #define above for max-object-size-bytes or something like that? I suppose the limiting factor will be the client certificate.

Reply:
Yes, good proposal. Will be fixed.